Automattic/pocket-casts-android
# Display most popular podcasts by category

## Description
- This PR displays the top 5 most popular podcasts by category that can be selected via category pill or categories bottom sheet modal

This is part of the categories redesign. You can find references and figma here in our internal reference: `pdeCcb-4tv-p2`.

> [!note]
> The recycler indicator view is not implemented on this PR

> [!note]
> The loading indicator is not implemented on this PR

## Testing Instructions
Have `CATEGORIES_REDESIGN` feature flag enabled
1. Click on some category pill
2. Make sure that 5 podcasts of the selected category are displayed below of category pill list menu
3. Change region
4. The categories filter must be reseted ( Remove the last selected category )
5. Open All categories
6. Click select a category from the categories bottom sheet dialog
7. Make sure that 5 podcasts of the selected category are displayed below of category pill list menu

## Screenshots or Screencast 
[Screen_recording_20240320_132802.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/3cb4edc1-2ebe-4252-bab0-f067520145dc)


## Checklist
 
#### I have tested any UI changes...
